db/migrate/20190604172236_create_widgets.rb

```diff
@@ -0,0 +1,12 @@
+class CreateWidgets < ActiveRecord::Migration[5.2]
+  def change
+    create_table :widgets do |t|
+      t.string :name, null: false
+      t.boolean :enabled, null: false
+      t.integer :duration_seconds, null: false
+      t.integer :order, null: false
```

Comment:
This is a bit of a nit, _but_ usually, at least in my experience with Rails that people call this field `position`, not `order`.  I think its better as well, because `order` is a description of a list, but `position` is the placement of a _single_ object in the list.  I'd rename.